tests/js/searchtools.js

```diff
@@ -1,16 +1,19 @@
 describe('Basic html theme search', function() {
 
+  function loadFixture(name) {
```

Comment:
I never used karma but should I assume that this is the way to do it? namely doing an XMLHttpRequest? isn't it possible just to like.. load the file normally?

Reply:
Good question.  So, two considerations: `karma` imports all configured JavaScript files when the tests run, and the `searchindex.js` files affect the state of the application (because it calls `setIndex`).

We want each test case to call `setIndex` individually -- setting state only for that test.

Doing that requires opening and `eval(...)`'ing the fixture.  The tests run in a browser, so they can't open local files -- and so we use the `karma` HTTP server to serve it up for access instead.

Reply:
Oh I see. Good then

Reply:
Thank you.